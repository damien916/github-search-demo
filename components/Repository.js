import React from 'react'
import Badge from 'material-ui/Badge'
import WatcherIcon from 'material-ui/svg-icons/action/visibility'
import StarIcon from 'material-ui/svg-icons/toggle/star'
import ForkIcon from 'material-ui/svg-icons/social/share'

const Repository = ({ name, url, language, forks, stars, watchers }) => {
  return (
    <div className="repository">
      <div className="repository__language">{language}</div>

      <div className="repository__infos">
        <div className="repository__name">{name}</div>
        <div className="repository__url">{url}</div>
      </div>

      <div className="repository__counts">
        <Badge badgeContent={watchers} secondary><WatcherIcon /></Badge>
        <Badge badgeContent={stars} secondary><StarIcon /></Badge>
        <Badge badgeContent={forks} secondary><ForkIcon /></Badge>
      </div>
    </div>
  )
}

export default Repository
