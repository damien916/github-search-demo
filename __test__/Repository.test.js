import React from 'react'
import renderer from 'react-test-renderer'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import Repository from '../components/Repository'

describe('Repository Snapshot Testing', () => {
  const component = renderer.create(
    <MuiThemeProvider>
      <Repository
        name="John Doe"
        url="http://www.john-doe.com"
        language="Javascript"
        forks={4}
        stars={5}
        watchers={6}
      />
    </MuiThemeProvider>
  )

  it('Render a repository', () => {
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})