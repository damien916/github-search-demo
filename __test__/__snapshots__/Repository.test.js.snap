// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository Snapshot Testing Render a repository 1`] = `
<div
  className="repository"
>
  <div
    className="repository__language"
  >
    Javascript
  </div>
  <div
    className="repository__infos"
  >
    <div
      className="repository__name"
    >
      John Doe
    </div>
    <div
      className="repository__url"
    >
      http://www.john-doe.com
    </div>
  </div>
  <div
    className="repository__counts"
  >
    <div
      style={
        Object {
          "display": "inline-block",
          "muiPrepared": true,
          "padding": "24px 24px 12px 12px",
          "position": "relative",
        }
      }
    >
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fill": "currentColor",
            "height": 24,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
        />
      </svg>
      <span
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#ff4081",
            "borderRadius": "50%",
            "color": "#ffffff",
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "fontSize": 12,
            "fontWeight": 500,
            "height": 24,
            "justifyContent": "center",
            "muiPrepared": true,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 24,
          }
        }
      >
        6
      </span>
    </div>
    <div
      style={
        Object {
          "display": "inline-block",
          "muiPrepared": true,
          "padding": "24px 24px 12px 12px",
          "position": "relative",
        }
      }
    >
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fill": "currentColor",
            "height": 24,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
      </svg>
      <span
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#ff4081",
            "borderRadius": "50%",
            "color": "#ffffff",
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "fontSize": 12,
            "fontWeight": 500,
            "height": 24,
            "justifyContent": "center",
            "muiPrepared": true,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 24,
          }
        }
      >
        5
      </span>
    </div>
    <div
      style={
        Object {
          "display": "inline-block",
          "muiPrepared": true,
          "padding": "24px 24px 12px 12px",
          "position": "relative",
        }
      }
    >
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fill": "currentColor",
            "height": 24,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
        />
      </svg>
      <span
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#ff4081",
            "borderRadius": "50%",
            "color": "#ffffff",
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "fontSize": 12,
            "fontWeight": 500,
            "height": 24,
            "justifyContent": "center",
            "muiPrepared": true,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 24,
          }
        }
      >
        4
      </span>
    </div>
  </div>
</div>
`;
